src/database/database_seeder.py

```diff
@@ -0,0 +1,270 @@
+        db.session.add(venue_obj)
+    db.session.commit()
+
+# Seed data for Course model
+def seed_courses():
+    courses = [
+        {'name': 'SWE71001', 'description': 'Software Engineering'},
+        {'name': 'SWE71002', 'description': 'Advance Software Development'},
+        {'name': 'SWE71002', 'description': 'DevOps'}
+    ]
+    for course in courses:
+        course_obj = Course(
+            name=course['name'],
+            description=course['description'],
+            created_at=datetime.now()
+        )
+        db.session.add(course_obj)
+    db.session.commit()
+
+# Seed data for LectureSchedule model
+def seed_lecture_schedules():
+    courses = Course.query.all()
+    semesters = Semester.query.all()
+    venues = Venue.query.all()
+
+    for _ in range(20):
+        course = fake.random_element(courses)
+        semester = fake.random_element(semesters)
+        venue = fake.random_element(venues)
+
+        start_time = fake.date_time_this_decade()
+        end_time = start_time + timedelta(hours=fake.random_element(elements=(1, 3)))
+
+        lecture_schedule = LectureSchedule(
+            course_id=course.id,
+            semester_id=semester.id,
+            venue_id=venue.id,
+            day=fake.random_element(elements=('Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday')),
+            start_time=start_time,
+            end_time=end_time,
+            one_off_date=fake.date_time_this_decade(),
+            is_recurring=fake.boolean(),
+            is_online=fake.boolean(),
+
+        )
+        db.session.add(lecture_schedule)
+
+    db.session.commit()
+
+# Seed data for user_roles association
+def seed_user_roles():
+    users = User.query.all()
+    roles = Role.query.all()
+
+    for user in users:
+        # Generate a random number of roles for each user
+        num_roles = fake.random_int(min=1, max=3)
+        user_roles = fake.random_elements(elements=roles, unique=True, length=num_roles)
+        user.roles.extend(user_roles)
+
+    db.session.commit()
+
+
+#logic for generating checkin code in the order "AA-BB-CC"
+def generate_check_in_code():
+    # Generate random uppercase characters for each segment of the check-in code
+    segment1 = fake.random_uppercase_letter() + fake.random_uppercase_letter()
+    segment2 = fake.random_uppercase_letter() + fake.random_uppercase_letter()
+    segment3 = fake.random_uppercase_letter() + fake.random_uppercase_letter()
+    check_in_code = f"{segment1}-{segment2}-{segment3}"
+    return check_in_code
+
+#Seed data for lecture_sessions association
+def seed_lecture_sessions():
+    # Query existing LectureSchedule objects to get their ids
+    lecture_schedules = LectureSchedule.query.all()
+    lecture_schedule_ids = [schedule.id for schedule in lecture_schedules]
+
+    for i in range(10):  # Create 10 lecture sessions
+        lecture_schedule_id = fake.random_element(elements=lecture_schedule_ids)
+        start_time = fake.date_time_this_decade()
```

Comment:
start_time and end_time is by default derived from the lecture_schedule. 

the reason for the seeming redundancy is for flexibility of lecture sessions (so a particular lecture session can be rescheduled, but not the entire parent lecture schedule)

Reply:
same goes for end_time